src/aoc2020/day1.clj

```diff
@@ -0,0 +1,143 @@
+(ns aoc2020.day1
+  (:require [aoc2020.util :refer [file->seq]]))
+
+
+;; part 1
+
+(defn strs->ints [strs]
+  "parsing strings to integers"
+  (let [str->int (fn [str]
+                   (Integer/parseInt str))]
+    (map str->int strs)))
+
+(defn int->compl [n sum]
+  "calculate complement"
+  (- sum n))
+
+(defn compl-in-set? [n sum s]
+  "detect whether the complement is in the sequence (using set)"
+  (let [compl (int->compl n sum)]                           ; calculate complement
+    (contains? s compl)))                                   ; check if exist in set
+
+(defn int->int*compl [n sum]
+  "multiply x and its complement"
+  (if n                                                     ; if n is not nil
+    (let [compl (int->compl n sum)]                         ; calculate complement
+      (* n compl))                                          ; multiply num and its complement
+    nil))
+
+(defn two-sum [sum strings]
+  "filter x for which both x and its complement are in the sequence
+  using set: complement-in-set?
+  using map: complement-in-map?"
+  (let [nums (strs->ints strings)                           ; parse strings
```

Comment:
The fact that you have to put a comment here to "parse strings" should tell you that the function name is insufficient.

Reply:
> The fact that you have to put a comment here to "parse strings" should tell you that the function name is insufficient.

yes, `strs->ints` is not a good name